xds/internal/balancer/clusterresolver/e2e_test/eds_impl_test.go

```diff
@@ -1138,3 +1143,168 @@ func waitForProducedZeroAddressesError(ctx context.Context, t *testing.T, client
 	}
 	return errors.New("timeout when waiting for RPCs to fail with UNAVAILABLE status and produced zero addresses")
 }
+
+// Test runs a server which listens on multiple ports. The test updates xds resouce
+// cache to contain a single endpoint with multiple addresses. The test intercepts
+// the resolver updates sent to the petiole policy and verifies that the
+// additional endpoint addresses are correctly propagated.
+func (s) TestEDS_EndpointWithMultipleAddresses(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	// Start a backend server which provide that listens to multiple ports to simulate
+	// a backend with multiple addresses.
+	servers, cleanup2 := startTestServiceBackends(t, 1)
+	defer cleanup2()
+	lis1, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Failed to create listener: %v", err)
+	}
+	defer lis1.Close()
+	lis2, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Failed to create listener: %v", err)
+	}
+	defer lis2.Close()
+
+	go servers[0].S.Serve(lis1)
+	go servers[0].S.Serve(lis2)
+
+	addrs, ports := backendAddressesAndPorts(t, servers)
+	additionalPorts := []uint32{
+		testutils.ParsePort(t, lis1.Addr().String()),
+		testutils.ParsePort(t, lis2.Addr().String()),
+	}
+
+	testCases := []struct {
+		name                      string
+		dualstackEndpointsEnabled bool
+		wantEndpointPorts         []uint32
+		wantAddrPorts             []uint32
+	}{
+		{
+			name:                      "flag_enabled",
```

Comment:
Nit: replace the underscore in the name with a space to make it more readable, and `t.Run` anyways does the opposite.

Reply:
TIL. Thanks.